File: SolariX.Api/DTOs/StationDtos.cs
// ============================================================================
// File: StationDtos.cs
// Project: SolariX - Smart Solar Microgrid Trading System
// Description: Data transfer objects for Solar Station / Microgrid Hub operations.
// Module: SE4040 Enterprise Application Development
// ============================================================================

using System.ComponentModel.DataAnnotations;

namespace SolariX.Api.DTOs
{
    public class CreateStationRequest
    {
        [Required]
        public string StationCode { get; set; } = string.Empty;

        [Required]
        public string StationName { get; set; } = string.Empty;

        [Required]
        public string LocationName { get; set; } = string.Empty;

        [Required]
        public double Latitude { get; set; }

        [Required]
        public double Longitude { get; set; }

        [Range(1, 100000)]
        public double CapacityKWh { get; set; }

        [Range(1, 500)]
        public int TotalBatterySlots { get; set; }

        public string OperationalHours { get; set; } = "06:00 - 20:00";
        public string ContactNumber { get; set; } = string.Empty;
    }

    public class UpdateStationRequest
    {
        public string StationName { get; set; } = string.Empty;
        public string LocationName { get; set; } = string.Empty;
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public double CapacityKWh { get; set; }
        public int TotalBatterySlots { get; set; }
        public int AvailableBatterySlots { get; set; }
        public string OperationalHours { get; set; } = "06:00 - 20:00";
        public string ContactNumber { get; set; } = string.Empty;
        public bool? IsActive { get; set; }
    }

    public class StationResponse
    {
        public string Id { get; set; } = string.Empty;
        public string StationCode { get; set; } = string.Empty;
        public string StationName { get; set; } = string.Empty;
        public string LocationName { get; set; } = string.Empty;
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public double CapacityKWh { get; set; }
        public int TotalBatterySlots { get; set; }
        public int AvailableBatterySlots { get; set; }
        public bool IsActive { get; set; }
        public string OperationalHours { get; set; } = string.Empty;
        public string ContactNumber { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
    }
}

File: mobile-client/app/src/main/java/com/solarix/mobile/data/remote/dtos/StationDtos.kt
package com.solarix.mobile.data.remote.dtos

import com.google.gson.annotations.SerializedName

data class StationResponse(
    @SerializedName("id") val id: String,
    @SerializedName("stationCode") val stationCode: String,
    @SerializedName("stationName") val stationName: String,
    @SerializedName("locationName") val locationName: String,
    @SerializedName("latitude") val latitude: Double,
    @SerializedName("longitude") val longitude: Double,
    @SerializedName("capacityKWh") val capacityKWh: Double,
    @SerializedName("totalBatterySlots") val totalBatterySlots: Int,
    @SerializedName("availableBatterySlots") val availableBatterySlots: Int,
    @SerializedName("isActive") val isActive: Boolean,
    @SerializedName("operationalHours") val operationalHours: String?,
    @SerializedName("contactNumber") val contactNumber: String?,
    @SerializedName("createdAt") val createdAt: String?
)

data class SlotResponse(
    @SerializedName("id") val id: String,
    @SerializedName("stationId") val stationId: String,
    @SerializedName("slotDate") val slotDate: String,
    @SerializedName("startTime") val startTime: String,
    @SerializedName("endTime") val endTime: String,
    @SerializedName("maxCapacityKW") val maxCapacityKW: Double,
    @SerializedName("availableCapacityKW") val availableCapacityKW: Double,
    @SerializedName("status") val status: String
)
